internal/util/prometheus.go

```diff
@@ -0,0 +1,74 @@
+package util
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type ControllerFunctionalPrometheusMetrics struct {
+	// ConfigCounter number of post /config to proxy successfully
+	ConfigCounter *prometheus.CounterVec
+
+	// ParseCounter number of ingress analysis failure
+	ParseCounter *prometheus.CounterVec
+
+	// ConfigureDurationHistogram duration of last successful confiuration sync
+	ConfigureDurationHistogram prometheus.Histogram
+}
+
+type Success string
```

Comment:
Please add documentation comments to all exported names.

Reply:
👍  will do.

Reply:
changed d0d14553f8bacff143d3319560b102f080d1022a